Dockerfile

```
@@ -1,10 +1,14 @@
-# Build the manager binary
-FROM ghcr.io/cybozu/golang:1.23-jammy as builder
+# Build the manager binary. This always executes in the native architecture of the building machine.
+FROM --platform=$BUILDPLATFORM ghcr.io/cybozu/golang:1.23-jammy AS builder
 
 COPY ./ .
-RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o accurate-controller ./cmd/accurate-controller
 
-# the controller image
+# Build the binary, cross-compiling if necessary
+ARG TARGETPLATFORM
+RUN CGO_ENABLED=0 GOOS=linux GOOARCH=$TARGETPLATFORM \
```

```suggestion
RUN CGO_ENABLED=0 GOOS=linux GOARCH=$TARGETARCH \
```

`TARGETPLATFORM` contains the target os.
Please use `TARGETARCH` instead.

---

When using TARGETPLATFORM, I got the following error.
```
go: unsupported GOOS/GOARCH pair linux/linux/amd64
```

---

oh goodness, yes this would not have worked. I did test the image but could only do so with docker's auto-binfmt thing, I'm guessing the x64 binary in the arm64 image just ran without emulation and I was none the wiser! Thanks for detailed checks on this :)